mmcv/cnn/bricks/transformer.py

```diff
@@ -0,0 +1,721 @@
+import torch
+import torch.nn as nn
+
+from ..utils import xavier_init
+from .activation import build_activation_layer
+from .norm import build_norm_layer
+from .wrappers import Linear
+
+
+class MultiheadAttention(nn.Module):
+    """A warpper for torch.nn.MultiheadAttention.
+
+    This module implements MultiheadAttention with residual connection.
```

Comment:
Is this module general enough?

Reply:
Hi, it implements self attention and multihead attention together, and plugins residual connection to format forwad_pre and forward_post in official DETR together.